In [2]:
import numpy as np
import pandas as pd

In [3]:
# The statistical module used to run chi square test
import scipy.stats as stats

In [4]:
# Observed data in a (hypothetical) survey of 6000 people 
observed = pd.Series([1500, 1350, 1600, 1550], index=["1", "2", "3", "4"])
observed.sum()/len(observed)

1500.0

In [6]:
# Create a data frame
df = pd.DataFrame([observed]).T

In [7]:
# Add a column whose default values are the expected values (sum up the series/number in the series)

df[1] = 1500

In [8]:
# Rename the data frame columns
df.columns = ["observed", "expected"]

In [9]:
# View the data frame
df

observed  expected
1      1500      1500
2      1350      1500
3      1600      1500
4      1550      1500

In [12]:
# Calculate the critical value
#The degree of freedom (df) is 4-1 = 3 (here they used number of rows)
# With a p-value of 0.05, the confidence level (q) is 1.00-0.05 = 0.95.
critical_value = stats.chi2.ppf(q = 0.95, df = 3)

In [13]:
# Print the critical value
critical_value

7.814727903251179

In [15]:
# Run the chi square test with stats.chisquare()
stats.chisquare(df.observed, df.expected)

Power_divergenceResult(statistic=23.333333333333336, pvalue=3.4411276663397364e-05)

## Since the Chi Squared is 23.333 and greater than Critical Value of 7.814, we conclude the results are statistically significant.  We can reject the null hypothesis
